src/reducers/reducer.js

```diff
@@ -1,51 +1,132 @@
+  return storage.has(key) ? storage.get(key) : []
+}
+
+// Persist the given tasks under the given key in the JSON storage file.
+function writeChanges(tasks, key) {
+  const existingTasks = getSavedTasks(key)
+  const newTasks = tasks.map(task => taskKey(task))
+  const allTasks = []
+  existingTasks.concat(newTasks).forEach(str => {
+    if (allTasks.indexOf(str) < 0) {
+      allTasks.push(str)
+    }
+  })
+  storage.set(key, allTasks)
+}
+
+function updateTasks(tasks, action) {
+  const tasksById = {}
+  const snoozedTasks = getSavedTasks(SNOOZED_KEY)
+  const archivedTasks = getSavedTasks(ARCHIVED_KEY)
+
+  // Add the existing tasks
+  tasks.forEach(task => (tasksById[task.id] = task))
 
-      // Add the existing tasks
-      tasks.forEach(task => (tasksById[task.id] = task))
+  // Update tasks with new values and add new tasks
+  action.tasks.forEach(task => {
+    tasksById[task.id] = Object.assign({}, tasksById[task.id], task)
+  })
 
-      // Update tasks with new values and add new tasks
-      action.tasks.forEach(task => {
-        tasksById[task.id] = Object.assign({}, tasksById[task.id], task)
-      })
+  const updatedTasks = Object.keys(tasksById).map(taskId => {
+    const task = tasksById[taskId]
+    const key = taskKey(task)
+    if (snoozedTasks.indexOf(key) > -1) {
+      task.snooze = true
+    }
+    if (archivedTasks.indexOf(key) > -1) {
+      task.archivedAt = storage.get(key)
+    }
+    return task
+  }).sort((a, b) => b.updatedAt - a.updatedAt) // Sort by updatedAt DESC
 
-      const updatedTasks = Object.keys(tasksById)
-        .map(taskId => tasksById[taskId])
-        .sort((a, b) => b.updatedAt - a.updatedAt) // Sort by updatedAt DESC
+  return updatedTasks
+}
+
+function selectTasks(tasks, action) {
+  return tasks.map(task => {
+    if (task.id === action.task.id) {
+      return Object.assign({}, task, { isSelected: true })
+    }
+    return task
+  })
+}
+
+function deselectTasks(tasks, action) {
+  return tasks.map(task => {
+    if (task.id === action.task.id) {
+      return Object.assign({}, task, { isSelected: false })
+    }
+    return task
+  })
+}
+
+function snoozeTasks(tasks) {
+  const snoozedTasks = []
+  const updatedTasks = tasks.map(task => {
+    if (task.isSelected) {
+      snoozedTasks.push(task)
+      return Object.assign({}, task, { snooze: true })
+    }
+    return task
+  })
+  writeChanges(snoozedTasks, SNOOZED_KEY)
+  return updatedTasks
+}
 
-      return updatedTasks
+function archiveTasks(tasks) {
+  const archivedTasks = []
+  const updatedTasks = tasks.map(task => {
+    if (task.isSelected) {
+      const date = new Date()
+      const archivedAt = date.toISOString()
+      storage.set(taskKey(task), archivedAt)
+      archivedTasks.push(task)
+      return Object.assign({}, task, { archivedAt })
+    }
+    return task
+  })
+  writeChanges(archivedTasks, ARCHIVED_KEY)
+  return updatedTasks
+}
+
+function tasksReducer(tasks = [], action) {
```

Comment:
Renamed the function to avoid a warning about the `tasks` parameter having the same name as the `tasks` function.